modules/publication-service-oaipmh/src/main/java/org/opencastproject/publication/oaipmh/OaiPmhPublicationServiceImpl.java

```diff
@@ -79,453 +80,648 @@
-
-  /** List of available operations on jobs */
-  private enum Operation {
-    Publish, Retract
+  public enum Operation {
+    Publish, Retract, UpdateMetadata
   }
 
-  /** The remote service registry */
-  private ServiceRegistry serviceRegistry = null;
-
-  /** The security service */
-  private SecurityService securityService = null;
-
-  /** The user directory service */
-  private UserDirectoryService userDirectoryService = null;
-
-  /** The organization directory service */
-  private OrganizationDirectoryService organizationDirectoryService = null;
-
-  /** The download distribution service */
-  private DownloadDistributionService downloadDistributionService = null;
-
-  /** The streaming distribution service */
-  private StreamingDistributionService streamingDistributionService = null;
-
-  /** The OAI-PMH persistence */
-  private OaiPmhDatabase persistence = null;
-
+  private DownloadDistributionService downloadDistributionService;
   private OaiPmhServerInfo oaiPmhServerInfo;
+  private OaiPmhDatabase oaiPmhDatabase;
+  private OrganizationDirectoryService organizationDirectoryService;
+  private SecurityService securityService;
+  private ServiceRegistry serviceRegistry;
+  private StreamingDistributionService streamingDistributionService;
+  private UserDirectoryService userDirectoryService;
 
-  /**
-   * Callback for the OSGi environment to set the service registry reference.
-   *
-   * @param serviceRegistry
-   *          the service registry
-   */
-  protected void setServiceRegistry(ServiceRegistry serviceRegistry) {
-    this.serviceRegistry = serviceRegistry;
-  }
-
-  /**
-   * Callback for setting the security service.
-   *
-   * @param securityService
-   *          the securityService to set
-   */
-  public void setSecurityService(SecurityService securityService) {
-    this.securityService = securityService;
-  }
-
-  /**
-   * Callback for setting the user directory service.
-   *
-   * @param userDirectoryService
-   *          the userDirectoryService to set
-   */
-  public void setUserDirectoryService(UserDirectoryService userDirectoryService) {
-    this.userDirectoryService = userDirectoryService;
+  public OaiPmhPublicationServiceImpl() {
+    super(JOB_TYPE);
   }
 
-  /**
-   * Sets a reference to the organization directory service.
-   *
-   * @param organizationDirectory
-   *          the organization directory
-   */
-  public void setOrganizationDirectoryService(OrganizationDirectoryService organizationDirectory) {
-    this.organizationDirectoryService = organizationDirectory;
+  @Override
+  protected String process(Job job) throws Exception {
+    if (!StringUtils.equalsIgnoreCase(JOB_TYPE, job.getJobType()))
+      throw new IllegalArgumentException("Can not handle job type " + job.getJobType());
+
+    MediaPackage mediaPackage = MediaPackageParser.getFromXml(job.getArguments().get(0));
+    Publication publication = null;
+    boolean checkAvailability = false;
```

Comment:
The use of variables in the case statement below is quite inconsistent: In the case UpdateMetadata, for example, the arguments are first stored into the variables and then used for the method call. In the case "Publish", the arguments are not first stored although the code does quite a similar thing.
"checkAvailability" could also be used in the case "Publish".

My preference would be:
* Always store the parsed value in a variable that is local to the case because that actual method call gets easier to read
* Duplicate variable declaration across cases instead of putting them into different scopes

I'm not expecting you to follow my personal preference, but please ensure that the code across the individual cases follows the same style.

Reply:
BTW: In case of always storing the parsed value into a meaningful variable, it would be ok to use integer literals for argument index as that would make the code understable.